nxdrive/gui/view.py

```diff
@@ -382,13 +421,35 @@ def rowCount(self, parent: QModelIndex = QModelIndex()) -> int:
     def set_sessions(
         self, sessions: List[Dict[str, Any]], parent: QModelIndex = QModelIndex()
     ) -> None:
-        self.beginRemoveRows(parent, 0, self.rowCount() - 1)
-        self.sessions.clear()
-        self.endRemoveRows()
+        if sessions and not self.shadow_session:
+            # Copy the first element from sessions and use it as shadow_session
+            self.shadow_session = sessions[0].copy()
+            self.shadow_session["shadow"] = True
+
+        # Create the sessions list with real datas from sessions.
+        for n_session in sessions:
+            self.add_session(parent, n_session)
+
+        # Add shadow_items to complete sessions list up to limit
+        while len(self.sessions) < 10:  # Limit the sessions list size to 10
```

Comment:
On the other hand, as we discussed OTR, this hardcoded value will break if one has more than 10 active sessions.
We need to find another way to handle that random (to be continued OTR).